File: backend/app/schemas/discovery.py
from datetime import date, datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class SearchResult(BaseModel):
    type: str
    id: UUID
    title: str
    snippet: str
    date: date | None
    metadata: dict[str, Any] = {}


class SearchResponse(BaseModel):
    items: list[SearchResult]
    page: int
    page_size: int
    total: int


class TimelineEntry(BaseModel):
    type: str
    id: UUID
    title: str
    occurred_at: datetime
    metadata: dict[str, Any] = {}


class TimelineResponse(BaseModel):
    items: list[TimelineEntry]
    page: int
    page_size: int
    total: int


class StatsSummary(BaseModel):
    reviews_completed_this_month: int
    current_review_streak: int
    longest_review_streak: int
    active_goals: int
    completed_goals: int
    goal_tasks_completed: int
    goal_tasks_total: int

File: frontend/src/types/discovery.ts
export interface SearchResult {
  type: 'review' | 'section' | 'preset' | 'goal' | 'task'
  id: string
  title: string
  snippet: string
  date: string | null
  metadata: Record<string, string>
}

export interface TimelineEntry {
  type: string
  id: string
  title: string
  occurred_at: string
  metadata: Record<string, string | number>
}

export interface StatsSummary {
  reviews_completed_this_month: number
  current_review_streak: number
  longest_review_streak: number
  active_goals: number
  completed_goals: number
  goal_tasks_completed: number
  goal_tasks_total: number
}

